src/react/TitleProvider.tsx

```diff
@@ -0,0 +1,140 @@
+import { useEffect, useState } from 'react'
+import { BotEvents } from 'mineflayer'
+import type { ClientOnMap } from '../generatedServerPackets'
+import Title from './Title'
+import type { AnimationTimes } from './Title'
+
+
+const defaultText: Record<string, any> = { 'text': '' }
+
+export default () => {
+  const [title, setTitle] = useState<string | Record<string, any>>(defaultText)
+  const [subtitle, setSubtitle] = useState<string | Record<string, any>>(defaultText)
+  const [actionBar, setActionBar] = useState<string | Record<string, any>>(defaultText)
+  const [animTimes, setAnimTimes] = useState<AnimationTimes>({ fadeIn: 400, stay: 3800, fadeOut: 800 })
+  const [open, setOpen] = useState(false)
+
+
+  useEffect(() => {
+    bot._client.on('set_title_text', (packet) => {
+      setTitle(JSON.parse(packet.text))
+      if (!open) {
+        setOpen(true)
+      }
+    })
+    bot._client.on('set_title_subtitle', (packet) => {
+      setSubtitle(JSON.parse(packet.text))
+    })
+    bot._client.on('action_bar', (packet) => {
+      setActionBar(JSON.parse(packet.text))
+      if (!open) {
```

Comment:
1. It always will be `false` since the effect does not update (empty dependency array).
2. No need for this if at all
3. Show state for `action bar` and `title + subtitle` are managed independently. So there should be two independent states

eg its possible that action bar closes earlier than title with subtitle

Reply:
Check last commit please